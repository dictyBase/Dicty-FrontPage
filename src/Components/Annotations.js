// @flow
import React from "react"
import { Box} from "grid-styled"
import FontAwsome from "react-fontawesome"
import styled from "styled-components"

import {
  AnnotationTitle,
  AnnotationSubTitle,
  AnnotationHeader,
  ListBox,
  AnnotationListItems,
  AnnotationContainer,
  MoreLink
} from "./styles"


const Container = styled.div`
display: flex;
flex-direction: row;
row-wrap: wrap;
width: 90%;
margin: auto;
`


const Annotations = (props) => {
  const genelist = props.annotations.genes.map((gene, index) =>
    <AnnotationListItems key={index}>
        {gene}
    </AnnotationListItems>,
  )

  const paperlist = props.annotations.papers.map((paper, index) =>
    <AnnotationListItems key={index}>
        {paper}
    </AnnotationListItems>,
  )

  return (
      <AnnotationContainer>
            <AnnotationHeader>
                <FontAwsome name="pencil fa-md" />
                <AnnotationTitle>
                  RECENT ANNOTATIONS
                </AnnotationTitle>
            </AnnotationHeader>
            <Container>
                <Box px={2} py={1} mt={-2} width={1 / 2}>
                    <AnnotationSubTitle>
                      Genes
                    </AnnotationSubTitle>
                    <ListBox margintop="0px" padbottom="0px">
                      {genelist}
                    </ListBox>  
                    <MoreLink padbottom="0px">
                      <FontAwsome name="plus fa-xs" />
                    </MoreLink>         
                </Box> 
                <Box px={2} py={1} mt={-2} width={1 / 2}>
                    <AnnotationSubTitle>
                      Papers
                    </AnnotationSubTitle>                
                    <ListBox margintop="0px" padbottom="0px">
                      {paperlist}
                    </ListBox> 
                    <MoreLink padbottom="0px">
                      <FontAwsome name="plus fa-xs" />
                    </MoreLink>                     
                </Box> 
            </Container>
      </AnnotationContainer>
  )
}

export default Annotations
