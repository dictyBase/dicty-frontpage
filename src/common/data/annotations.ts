export default {
  genes: ["erkB", "kif8", "cln3", "tpp1"],
  papers: ["30612939", "31067156", "30771446", "31100984"],
}
