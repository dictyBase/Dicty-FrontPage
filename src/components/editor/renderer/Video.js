import React, { PureComponent } from "react"

export default class Video extends PureComponent {
  onClick = e => {
    e.stopPropagation()
  }

  render() {
    const { node, isSelected, editor, attributes } = this.props
    const src = node.data.get("src")
    const url = node.data.get("url")

    const { readOnly } = editor.props

    const wrapperStyle = {
      position: "relative",
      paddingBottom: "50.66%",
      paddingTop: "25px",
      height: "0",
      outline: isSelected ? `2px solid #0017ff` : "none",
    }

    const maskStyle = {
      display: isSelected ? "none" : "block",
      position: "absolute",
      top: "0px",
      left: "0px",
      right: "0px",
      bottom: "0px",
      height: "100%",
      cursor: "cell",
      zIndex: 1,
    }

    const iframeStyle = {
      position: "absolute",
      top: "0px",
      left: "0px",
      width: "100%",
      height: "100%",
    }

    return (
      <div>
        <div {...attributes} style={wrapperStyle}>
          <div style={readOnly ? {} : maskStyle} />
          <iframe
            title="video-embed"
            id="ytplayer"
            type="text/html"
            width="580"
            height="390"
            src={src}
            frameBorder="0"
            style={iframeStyle}
          />
        </div>
      </div>
    )
  }
}
