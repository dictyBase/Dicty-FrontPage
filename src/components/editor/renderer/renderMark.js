// @flow
import React, { type Node } from "react"
import { type Mark } from "slate"
import MARKS from "components/editor/constants/marks"

type Props = {
  children: Node,
  mark: Mark,
}

const renderMark = (props: Props) => {
  const { children, mark } = props
  switch (mark.type) {
    case MARKS.BOLD:
      return <strong>{children}</strong>

    case MARKS.ITALIC:
      return <em>{children}</em>

    case MARKS.UNDERLINE:
      return <u>{children}</u>

    case MARKS.STRIKETHROUGH:
      return <del>{children}</del>

    default:
      return null
  }
}

export default renderMark
