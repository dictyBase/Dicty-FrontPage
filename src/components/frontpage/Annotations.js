// @flow
import React from "react"
import { withStyles } from "@material-ui/core/styles"
import Grid from "@material-ui/core/Grid"
import FontAwesome from "react-fontawesome"

import {
  AnnotationTitle,
  AnnotationSubTitle,
  AnnotationHeader,
  ListBox,
  AnnotationListItems,
  AnnotationContainer,
  MoreLink,
} from "styles"

const styles = theme => ({
  container: {
    margin: "auto",
  },
  box: {
    padding: "1px 2px 1px 2px",
    marginTop: "-2px",
  },
})

type Props = {
  /** The annotations data in object form */
  annotations: Object,
  /** Material-UI styling */
  classes: Object,
}

/** Widget that displays the most recent annotations for genes and papers */

const Annotations = (props: Props) => {
  const { annotations, classes } = props

  const genelist = annotations.genes.map((gene, index) => (
    <AnnotationListItems key={index}>{gene}</AnnotationListItems>
  ))

  const paperlist = annotations.papers.map((paper, index) => (
    <AnnotationListItems key={index}>{paper}</AnnotationListItems>
  ))

  return (
    <AnnotationContainer>
      <AnnotationHeader>
        <FontAwesome name="pencil fa-md" />
        <AnnotationTitle>RECENT ANNOTATIONS</AnnotationTitle>
      </AnnotationHeader>
      <Grid container className={classes.container} wrap="wrap">
        <Grid item className={classes.box} xs={6}>
          <AnnotationSubTitle>Genes</AnnotationSubTitle>
          <ListBox margintop="0px" padbottom="0px">
            {genelist}
          </ListBox>
          <MoreLink padbottom="0px">
            <FontAwesome name="plus fa-xs" />
          </MoreLink>
        </Grid>
        <Grid item className={classes.box} xs={6}>
          <AnnotationSubTitle>Papers</AnnotationSubTitle>
          <ListBox margintop="0px" padbottom="0px">
            {paperlist}
          </ListBox>
          <MoreLink padbottom="0px">
            <FontAwesome name="plus fa-xs" />
          </MoreLink>
        </Grid>
      </Grid>
    </AnnotationContainer>
  )
}

export default withStyles(styles)(Annotations)
