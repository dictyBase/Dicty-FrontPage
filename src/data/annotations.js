export default {
    genes: ["abprtA", "trgrB1", "tgrC1", "tacA"],
    papers: ["25609090", "25546705", "25540127", "25596489"]
}