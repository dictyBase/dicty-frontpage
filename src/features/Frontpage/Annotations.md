```js
import annotations from "data/annotations"

;<Annotations annotations={annotations} />
```
