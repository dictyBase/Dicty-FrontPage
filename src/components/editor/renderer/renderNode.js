// @flow
import React, { type Node } from "react"
import { type Block } from "slate"
import styled from "styled-components"
import Video from "components/editor/renderer/Video"
import BLOCKS from "components/editor/constants/blocks"
import INLINES from "components/editor/constants/inlines"

type Props = {
  node: Block,
  children: Node,
  attributes: Object,
}

const Image = styled.img`
  display: block;
  max-width: 100%;
  max-height: 20em;
  box-shadow: ${props => (props.selected ? "0 0 0 2px blue;" : "none")};
`

const renderNode = (props: Props) => {
  const { attributes, children, node } = props
  switch (node.type) {
    case BLOCKS.UL_LIST:
      return <ul {...attributes}>{children}</ul>

    case BLOCKS.OL_LIST:
      return <ol {...attributes}>{children}</ol>

    case BLOCKS.LIST_ITEM:
      return <li {...attributes}>{children}</li>

    case BLOCKS.HEADING_1:
      return <h1 {...attributes}>{children}</h1>

    case BLOCKS.HEADING_2:
      return <h2 {...attributes}>{children}</h2>

    case BLOCKS.HEADING_3:
      return <h3 {...attributes}>{children}</h3>

    case BLOCKS.BLOCKQUOTE:
      return <blockquote {...attributes}>{children}</blockquote>

    case INLINES.LINK: {
      const { data } = node
      const href = data.get("href")
      return (
        <a {...attributes} href={href}>
          {children}
        </a>
      )
    }

    case BLOCKS.IMAGE: {
      const { data, isSelected } = node
      const src = data.get("src")
      const className = isSelected ? "selected" : "unselected"
      const style = { display: "inline-block" }
      return (
        <img
          src={src}
          className={className}
          {...attributes}
          style={style}
          alt=""
        />
      )
    }

    case BLOCKS.ALIGN_LEFT:
      return (
        <div style={{ textAlign: "left" }} {...attributes}>
          {children}
        </div>
      )

    case BLOCKS.ALIGN_CENTER:
      return (
        <div style={{ textAlign: "center" }} {...attributes}>
          {children}
        </div>
      )

    case BLOCKS.ALIGN_RIGHT:
      return (
        <div style={{ textAlign: "right" }} {...attributes}>
          {children}
        </div>
      )

    case BLOCKS.HR:
      return <hr {...attributes} />

    case BLOCKS.TABLE:
      return (
        <table {...attributes}>
          <tbody>{children}</tbody>
        </table>
      )

    case BLOCKS.TABLE_ROW:
      return <tr {...attributes}>{children}</tr>

    case BLOCKS.TABLE_CELL:
      return <td {...attributes}>{children}</td>

    case BLOCKS.VIDEO:
      return <Video {...props} />

    default:
      return null
  }
}

export default renderNode
