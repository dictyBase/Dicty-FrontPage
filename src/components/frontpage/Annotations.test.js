import React from "react"
import renderer from "react-test-renderer"
import Annotations from "components/frontpage/Annotations"
import annotations from "data/annotations"
import "jest-styled-components"

test("Annotations snapshot test", () => {
  const component = renderer.create(<Annotations annotations={annotations} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
