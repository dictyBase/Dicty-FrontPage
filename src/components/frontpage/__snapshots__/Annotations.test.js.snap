// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Annotations snapshot test 1`] = `
.c4 {
  box-sizing: border-box;
  width: 50%;
  margin-top: -16px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
}

.c0 {
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  line-height: 1.428;
  text-align: center;
  color: #084b8a;
  background-color: #eff8fb;
  padding: 5px 5px 0 5px;
  border-top-left-radius: 10px;
  border-top-right-radius: 10px;
  margin-bottom: 10px;
}

.c1 {
  color: #0489b1;
  font-size: 16px;
  border-top-left-radius: 5px;
  border-top-right-radius: 5px;
  background-color: #fff;
  vertical-align: middle;
  text-align: center;
  padding: 10px 0 10px 0;
  margin-bottom: 10px;
}

.c2 {
  padding-left: 5px;
  font-weight: bold;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  row-wrap: wrap;
  width: 90%;
  margin: auto;
}

.c5 {
  padding-top: 5px;
  font-size: 14px;
  font-weight: bold;
}

.c6 {
  padding: 0px 25px 10px 25px;
  font-size: 12px;
  margin-bottom: 5px;
  padding-bottom: 0px;
  margin-top: 0px;
}

.c7 {
  list-style-type: none;
  font-size: 11px;
}

.c8 {
  color: #0b3861;
  font-size: 11px;
  font-style: italic;
  font-weight: normal;
  text-align: center;
  padding-bottom: 0px;
}

@media (max-width:767px) {
  .c1 {
    font-size: 18px;
    text-align: center;
  }
}

@media (max-width:992px) and (min-width:767px) {
  .c6 {
    font-size: 10px;
  }
}

@media (max-width:768px) {
  .c6 {
    font-size: 16px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span
      aria-hidden={true}
      className="fa fa-pencil fa-md"
    />
    <span
      className="c2"
    >
      RECENT ANNOTATIONS
    </span>
  </div>
  <div
    className="c3"
  >
    <div
      className="c4"
      is={null}
    >
      <span
        className="c5"
      >
        Genes
      </span>
      <ul
        className="c6"
      >
        <li
          className="c7"
        >
          abprtA
        </li>
        <li
          className="c7"
        >
          trgrB1
        </li>
        <li
          className="c7"
        >
          tgrC1
        </li>
        <li
          className="c7"
        >
          tacA
        </li>
      </ul>
      <div
        className="c8"
      >
        <span
          aria-hidden={true}
          className="fa fa-plus fa-xs"
        />
      </div>
    </div>
    <div
      className="c4"
      is={null}
    >
      <span
        className="c5"
      >
        Papers
      </span>
      <ul
        className="c6"
      >
        <li
          className="c7"
        >
          25609090
        </li>
        <li
          className="c7"
        >
          25546705
        </li>
        <li
          className="c7"
        >
          25540127
        </li>
        <li
          className="c7"
        >
          25596489
        </li>
      </ul>
      <div
        className="c8"
      >
        <span
          aria-hidden={true}
          className="fa fa-plus fa-xs"
        />
      </div>
    </div>
  </div>
</div>
`;
